controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -557,7 +557,7 @@ private void logIfEmpty(String delegationToken, String requestName, String scope
                         logAndUntrackRequestTag(requestTag);
                     });
         } catch (Exception e) {
-            log.error(e.getMessage(), e);
+            log.error("Encountered exception in authenticateExecuteAndProcessResults: {}", e.getMessage(), e);
             logAndUntrackRequestTag(requestTag);
             streamObserver.onError(Status.UNAUTHENTICATED
```

Comment:
As discussed, we should set the status as `Status.UNAUTHENTICATED` only incase of authentication errors, else the status should be `Status.INTERNAL`.  

The client does not retry in cases of Authentication failures ref: https://github.com/pravega/pravega/blob/26bfc25eca2e76e691bc32177ef2500cccb95879/client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java#L202-L218

Doing this change will ensure the client retries on non-auth related errors.

Reply:
Sure, thanks for spotting the issue. I've modified the code to set appropriate status: 
* `io.grpc.Status.UNAUTHENTICATED` upon authentication failures 
* `io.grpc.Status.PERMISSION_DENIED` upon authorization failures
* `io.grpc.Status.INTERNAL` upon internal failures.